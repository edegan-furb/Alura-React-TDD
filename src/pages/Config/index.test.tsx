import React from "react";
import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import Config from ".";

const mockNavegation = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavegation,
  };
});

describe("The config page", () => {
  test("Must be rendered correctly", () => {
    const { container } = render(
      <RecoilRoot>
        <Config />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
