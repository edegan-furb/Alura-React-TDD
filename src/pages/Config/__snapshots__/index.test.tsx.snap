// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The config page Must be rendered correctly 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Lets begin!
      </h2>
      <form>
        <div
          class="group-input-btn"
        >
          <input
            placeholder="Enter participant names"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Add
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="footer-config"
      >
        <button
          class="button"
          disabled=""
        >
          Start game
        </button>
        <img
          alt="Sacolas de compras"
          src="/images/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
